homeassistant/components/wilight/__init__.py

```diff
@@ -10,7 +10,7 @@
 from .parent_device import WiLightParent
 
 # List the platforms that you want to support.
-PLATFORMS = ["light"]
+PLATFORMS = ["light", "fan"]
```

Comment:
This list is used in config_flow.py as well. I think it is a good practice to put this in const.py.

Reply:
During the approval of PR # 36694 I tried to use only one constant, but Martin was incisive to have two constants. See below:

  @MartinHjelmare
MartinHjelmare 13 days ago  Member
Please create a separate constant for allowed_components. We want to keep the home assistant API separate from the Wilight API as either of those may change.
  @leofig-rj
leofig-rj 13 days ago  Author  Contributor
PLATFORMS is a constant that comes from INIT.py and not from the Wilight API.
It serves to indicate which components are allowed by the current integration.
I used allowed_components for your suggestion.
I used the name PLATFORMS, because it was already in a template created by Homeassistant.
Would you like me to rename PLATFORMS to ALLOWED_COMPONENTS in INIT.py?
  @leofig-rj
leofig-rj 13 days ago  Author  Contributor
This is where I see if the components available in the WiLight API are allowed by the current integration.
  @MartinHjelmare
MartinHjelmare 12 days ago  Member
The PLATFORMS constant contains strings for the home assistant api. allowed_components should contain strings for the wilight api. They must be created in different separate variables. We have to create a new constant, named eg ALLOWED_WILIGHT_COMPONENTS. Don't change PLATFORMS.
  @leofig-rj
leofig-rj 12 days ago  Author  Contributor
Honestly I don't see it like that, I see a check of the components of the WiLight API (self._components) against the components of the Homeassistant API (PLATFORMS), to see if the components of the WiLight API are allowed in this Homeassistant integration. self._components is updated by the _self_update function.
I am renaming self._components to self._wilight_components and _self_update to _wilight_update for clarity.
If you really want me to create a new constant, I will, but it doesn't make sense because I will have to update PLATFORMS and the new constant equally in an upcoming PR to include new components.
  @leofig-rj
leofig-rj 12 days ago  Author  Contributor
This conversation made me review the concept of this comparison and I saw that where any was, in reality it should be all.
In this way, only if ALL WiLight components are allowed for the HA integration version.
This occurs in the case of WiLight for ceiling fan, which has the light component and the fan on the same device.

     allowed_components = PLATFORMS
     component_ok = all (
         wilight_component in allowed_components
         for wilight_component in self._wilight_components
     )
  @MartinHjelmare
MartinHjelmare 11 days ago  Member
Please create separate constants.
  @leofig-rj
leofig-rj 11 days ago  Author  Contributor
Done

Reply:
Oh, I understand that. Didn't know they should be separated.

Reply:
For my part I would only have the constant PLATFORMS and it could be in const.py, but it is like that at Martin's recommendation.

Reply:
I don't agree with Martin's idea, but if he says 'do it', then do it. :D

Reply:
We don't want to directly couple two different APIs since if one would change we would break the integration without explicitly knowing that. API changes must always show explicitly.

Reply:
Hmm, it started to make sense now.